### DISCUSSION TITLE
Configuration Home Assistant Energy Panel

### DISCUSSION THREAD

**Discussion by metcrl** [hultenvp/solis-sensor#103] [Q&A]
Hi Guys,
I add the solis integration with success, and now i'm trying to configure the energy panel on Home Assistant.
My answer is: which sensor i have to add in grid (exported/imported) and in battery (charge/discharged)?

I tried to add the daily_grid_energy_purchased, daily_on_grid_energy, daily_energy_charged, daily_energy_discharged.
But all worked just for the first day; during the second day, the sensors continue to show the energy accumulated / lost in the previous 24 hours ruining the home assistant statisctics

**Comment by hultenvp** [comment]
It could be that those sensors suffer from the same issue we had with energy today: The stats the server provide don't reset at midnight, but in the morning when the inverter comes back online. 
I do not have a hybrid solis inverter myself. Could you post screenshots from the 48h sensor history of the sensors you mention here?
If it is the problem, then we can fix it.
I assume you are using m.ginlong.com portal, not SolisCloud?

Cheers

**Reply by metcrl** [reply]
Thanks for the reply!
Yes actually I'm using the old portal but I've already ask the api from the support to use SolisCloud.
I think that the problem is exactly this, the stats don't reset at midnight.
I attach the requested screenshot
![20220312_102010](https://user-images.githubusercontent.com/70216900/158012154-322ea851-3fd3-4df6-8ce2-141affa82a2d.jpg)

**Reply by hultenvp** [reply]
This is odd. 
First of all I'd expect a daily sensor to reset to 0 at preferably midnight, but this one keeps on increasing. It's however still a low kWh value, so either the inverter is new or it does reset at some point. The dashboard must be able to deal with these kind of graphs. It's unable to deal with spikes, but there aren't any. Can you share screenshots from the graphs on ginlong portal itself? I'd like to understand what's going on.
Second, I'd also expect discharge to continue after midnight. I assume your inverter does not stop at midnight discharging the battery if there's still demand? Can you share also a screenshot of the inverter state sensor? Same period if possible, so I can compare.

Cheers

**Reply by metcrl** [reply]
Hello, this is strange. Yesterday (March 13) all the sensor in HA seems to reset at midnight. In fact, the energy panel in HA yesterday worked perfectly.
Today morning I take a look and I see that, again, the sensors didn't reset.
I didn't do nothing different.
I'll take as example the battery discharge sensor, but the others sensor have the same behavior. I attach the screens.
![image](https://user-images.githubusercontent.com/70216900/158126115-025e2df4-e35b-4778-b047-7d8431f3b3b5.png)
![image](https://user-images.githubusercontent.com/70216900/158126423-4544d0b2-5bd6-4a08-b488-12591b834191.png)
![image](https://user-images.githubusercontent.com/70216900/158126493-6e2bd94b-67df-4340-b863-6d9fefc203d7.png)
![image](https://user-images.githubusercontent.com/70216900/158126553-f40edf59-db8a-47f8-87bc-f721256c6f0f.png)

And this is the energy panel in HA of the last 3 days. As you can see the only one that worked fine is March 13:
![image](https://user-images.githubusercontent.com/70216900/158126952-1caafb25-1c57-45d0-bba6-0093ec166ac0.png)
![image](https://user-images.githubusercontent.com/70216900/158126996-4d987d6b-7dc2-4b45-92e1-3c2b512bbff5.png)
![image](https://user-images.githubusercontent.com/70216900/158127031-77a81788-98c9-4af7-ad82-8ca54f786c03.png)

**Comment by hultenvp** [comment]
Perhaps it only resets once a week? Seems too much of a coincidence that it is Sunday. Other thing could be that the sensor data only resets if the inverter switches off. 
What does the data on ginlong.com look like?
And what does solis state sensor look like? Could you share the graph of solis state for the same period of 12-14 march?
I reviewed the code once more, key I'm reading out is the correct one for daily battery discharging, so that can't be the issue

**Comment by metcrl** [comment]
Hello! Today morning the sensors reset again at midnight...
![image](https://user-images.githubusercontent.com/70216900/158328752-2e64fea4-7bda-46d9-b2d5-5bf9ebbf6b1e.png)

So we can exclude that reset happen on Sunday. The inverter will never switch off, because the battery stop to discharge at 20% of capacity so all remain online 24/24.
What did you mean by data on ginlong.com? The soliscloud platform? Anyway the data on soliscloud start from 0 every day at midnight

The solis state is always at 1, because the inverter is always on
![image](https://user-images.githubusercontent.com/70216900/158328246-cb55580f-c84d-45f3-b9df-98b284cf7190.png)

**Reply by hultenvp** [reply]
The integration supports the old ginlong platform (m.ginlong.com) as well as the new soliscloud platform (soliscloud.com).
The old one uses username & password, the new one uses an API key.

I understood from your response earlier this thread that you are still on the old ginlong platform, hence my question what the graphs show if you login there and look at battery discharge history.

From the thread history I know understand the following:
* Your integration is configured to use m.ginlong.com (old platform)
* You requested API's, but AFAICT you are not yet using them
* In HA you see that (amongst others) daily battery discharge does not reset every night
* On the m.ginlong.com portal the graph does reset every night.

Is that correct or do I misunderstand something?
